the cleanup for GtkStackPage is defined in gtk4.0 headers themselves
at /usr/include/gtk-4.0/gtk/gtkstack.h:182

diff --git a/src/libide/tweaks/ide-tweaks-window.c b/src/libide/tweaks/ide-tweaks-window.c
index f2553a3..bb32acd 100644
--- a/src/libide/tweaks/ide-tweaks-window.c
+++ b/src/libide/tweaks/ide-tweaks-window.c
@@ -34,3 +34,2 @@
 
-G_DEFINE_AUTOPTR_CLEANUP_FUNC (GtkStackPage, g_object_unref)
 
